### ISSUE TITLE
header only compile mode is broken

### ISSUE THREAD

**Issue by Florian-Thake** [Florian-Thake/TeaScript-Cpp-Library#6]
If `Engine.hpp` or` CoroutineScriptEngine.hpp` is included in more than one translation unit (.cpp file) the linker will produce errors like
"error LNK2005: "public: virtual void __cdecl teascript::Engine::ResetState(void)" (?ResetState@Engine@teascript@@UEAAXXZ) already defined in main.obj"

Thank you to the user (L.M.) who was reporting this issue via email! :)

**Comment by Florian-Thake** [comment]
2 possible workarouds (the last is always possible):

You can either try to change your code so that Engine.hpp is included only in one .cpp file, or you need to switch to "non header only mode" like it is done in the teascript demo project from the GitHub repo.

For that you need to pick one .cpp file of your project in which all the TeaScript definitions shall go.
In that .cpp file you need to do these 2 defines:
```

#define TEASCRIPT_DISABLE_HEADER_ONLY       1
#define TEASCRIPT_INCLUDE_DEFINITIONS       1
```

before any TeaScript includes. After that you need to include Engine.hpp

In all other .cpp files of your project where you include TeaScript you must add this define prior any TeaScript include:
`#define TEASCRIPT_DISABLE_HEADER_ONLY       1`

**Comment by Florian-Thake** [comment]
This bug is fixed with commit 51977e66039249936dd8d34f71533ef17cfd763a
